### DISCUSSION TITLE
Help with building as a shared library

### DISCUSSION THREAD

**Discussion by cornzzy** [erebe/wstunnel#353] [Q&A]
Is it possible to build wstunnel for embedding purposes? I'm looking to build as .aar or .so (android), .xcframework (ios), .dll (windows).

Only [client](https://github.com/erebe/wstunnel/blob/main/src/tunnel/client/client.rs) functions which can run a client server. Would you please point me to the right direction.

**Comment by erebe** [comment]
For now, it will be difficult because the code in not split between a lib and a exec part. The build is doing only an executable.
So for it to be possible, there should be some refacto first in order to do this split

**Comment by jefcolbi** [comment]
@erebe Hi.
Coming to this and what to ask if it is already possible? I want to integrate it in a flutter app.
Can this [https://github.com/fzyzcjy/flutter_rust_bridge](https://github.com/fzyzcjy/flutter_rust_bridge) work ? 

Thanks

**Comment by erebe** [comment]
I have no idea regarding how to embed it in another app.

But `wstunnel` crate is a library so you can build it in a standalone way. It exposes only a few entrypoints to start either a server or a client 



https://github.com/erebe/wstunnel/blob/main/wstunnel/src/lib.rs#L41
